src/main/java/com/redhat/cloud/notifications/routers/UserConfigService.java

```diff
@@ -213,9 +221,14 @@
                                                 return Multi.createFrom().empty();
                                             }).concatenate();
                                 }).concatenate();
-                    }).concatenate().collect().asList().onItem().transformToMulti(objects -> emailSubscriptionResources.getEmailSubscriptionsForUser(account, username))
+                    }).collect().asList().onItem().transformToMulti(objects -> emailSubscriptionResources.getEmailSubscriptionsForUser(account, username))
                     .onItem().transform(emailSubscription -> {
-                        values.bundles.get(emailSubscription.getBundleName()).applications.get(emailSubscription.getApplicationName()).notifications.put(emailSubscription.getType(), true);
+                        if (values.bundles.containsKey(emailSubscription.getBundleName())) {
+                            BundleSettingsValue bundleSettingsValue = values.bundles.get(emailSubscription.getBundleName());
+                            if (bundleSettingsValue.applications.containsKey(emailSubscription.getApplicationName())) {
+                                bundleSettingsValue.applications.get(emailSubscription.getApplicationName()).notifications.put(emailSubscription.getType(), true);
+                            }
+                        }
```

Comment:
Things would be much simpler if we had SQL constraints between the `endpoint_email_subscriptions`, `bundles` and `applications` tables. We should take care of that soon.